File: distros/W/WebService-Eulerian-Analytics/other/website-getall.pl
#!/usr/bib/perl -w
#
# $Id: website-getall.pl,v 1.1 2008-09-03 18:46:02 cvscore Exp $

use strict;
use SOAP::Lite;
use Data::Dumper;

my $apikey	= 'YOUR_API_KEY';
my $host	= 'YOUR_API_HOST';

my $soap = SOAP::Lite->proxy( $host.'/ea/v1/Website' );

# header params for auth
my @a_hdr	= ( 
 SOAP::Header->name("apikey")->value( $apikey )->type('')
);

# additionnal params for method
my @a_p		= ();

my $result	= $soap->call(
   SOAP::Data->name('getAll')->uri('Website') => @a_hdr, @a_p );

if ( $result->fault ) {
 die $result->faultstring;
}

print Dumper($result->valueof('//getAllResponse/getAllReturn'));

1;
__END__

File: distros/W/WebService-Eulerian-Analytics/other/website-getall.php
<?php
/**
 * PHP : 5.2.6-3
 * PHP-SOAP : 0.11
 *
 * $Id: website-getall.php,v 1.1 2008-09-03 18:46:02 cvscore Exp $
 */

class soap_header {
 private $apikey;
 public function __construct ( $apikey ) {
  $this->apikey = $apikey;
 }
}

$host		= 'YOUR_API_HOST';
$apikey		= 'YOUR_API_KEY';
$login_hdr	= new soap_header( $apikey );
$login_hdr_v	= new SoapVar($login_hdr, SOAP_ENC_OBJECT);
$header		= new SoapHeader("EA", "SOAP-ENV", $login_hdr_v, false);

$soap 		= new SoapClient(null, array(
 location      => $host.'/ea/v1/Website',
 uri           => 'Website' )
);

// getAll
$result  	= $soap->__soapCall("getAll", array(), null, $header);

foreach ( $result as $hash ) {
 echo "--- website_name=".$hash->website_name."\n";
}

?>
